### PR TITLE
Fix duplicate method exception when registering ai services

### PR DESCRIPTION
**Pull request by iocanel**
Resolves: #491

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/MethodUtil.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package io.quarkiverse.langchain4j.deployment;
+
+import org.jboss.jandex.MethodInfo;
+
+public final class MethodUtil {
```
Comment (geoand) [inline core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/MethodUtil.java:5]: I am wondering if there is already a util like this in Quarkus

**Review by geoand — COMMENTED** [review]
Thanks for this!

I added a comment

File: core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/MethodUtil.java (hunk 1 above)
Comment (iocanel) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/MethodUtil.java:5]: Maybe `Methods` has something like it. Looking.

Comment (iocanel) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/MethodUtil.java:5]: Didn't find any.

Comment (geoand) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/MethodUtil.java:5]: I'll get back to you in a few minutes

Comment (iocanel) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/MethodUtil.java:5]: Ok, meanwhile I added a couple of tests:
- one the verifies the problem
- one the verifies the correctness of the util

Comment (geoand) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/MethodUtil.java:5]: > I'll get back to you in a few minutes

Does `io.quarkus.gizmo.SignatureBuilder` do the trick?

Comment (iocanel) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/MethodUtil.java:5]: Let me check

Comment (iocanel) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/MethodUtil.java:5]: No, I don't think we can use that.

They don't seem to provide the name and that makes it of limited use to our use case: https://github.com/quarkusio/gizmo/blob/main/src/main/java/io/quarkus/gizmo/MethodSignatureBuilderImpl.java

**Review by geoand — APPROVED** [review]
Thanks!

**Pull request merged**